File: viewpager/src/main/java/com/donfyy/viewpager/lazyloading/lazy3/Fragment2WithViewPager.java
package com.donfyy.viewpager.lazyloading.lazy3;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.donfyy.viewpager.R;

import java.util.ArrayList;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;


public class Fragment2WithViewPager extends LazyFragment2 {

    private static final String TAG = "Fragment2WithViewPager";

    private ViewPager viewPager;  //对应的viewPager
    private ArrayList<Fragment> fragmentsList;//view数组


    public static Fragment newIntance() {
        Fragment2WithViewPager fragment = new Fragment2WithViewPager();
        fragment.setFragmentDelegater(new FragmentDelegater(fragment));
        return fragment;
    }

    @Override
    protected int getLayoutRes() {
        return R.layout.fragment_2_vp;
    }

    @Override
    protected void initView(View view) {
        viewPager = view.findViewById(R.id.viewpager01);

        fragmentsList = new ArrayList<>();

        fragmentsList.add(Fragment2_vp_1.newIntance());
        fragmentsList.add(Fragment2_vp_2.newIntance());
        fragmentsList.add(Fragment2_vp_3.newIntance());
        fragmentsList.add(Fragment2_vp_4.newIntance());

        /**
         * 实例化一个PagerAdapter
         * 必须重写的两个方法
         * getCount
         * getItem
         */

        PagerAdapter pagerAdapter = new FragmentPagerAdapter(getChildFragmentManager()) {
            @Override
            public Fragment getItem(int i) {
                return fragmentsList.get(i);
            }

            @Override
            public int getCount() {
                return fragmentsList.size();
            }
        };

        viewPager.setAdapter(pagerAdapter);
    }

    @Override
    protected void onFragmentFirstVisible() {

    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return super.onCreateView(inflater, container,savedInstanceState);
    }



    @Override
    public void onResume() {
        super.onResume();
        Log.d(TAG, "onResume: " + "Fragment2WithViewPager");
    }

    @Override
    public void onPause() {
        super.onPause();
    }
}

File: viewpager/src/main/java/com/donfyy/viewpager/lazyloading/lazy4/Fragment2WithViewPager.kt
package com.donfyy.viewpager.lazyloading.lazy4

import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.donfyy.viewpager.R
import java.util.*

class Fragment2WithViewPager : BaseFragment() {
    private lateinit var viewPager //对应的viewPager
            : ViewPager
    private lateinit var fragmentsList //view数组
            : ArrayList<Fragment>

    override val layoutRes: Int
        get() = R.layout.fragment_2_vp

    override fun initView(view: View?) {
        super.initView(view)
        view?:return
        viewPager = view.findViewById(R.id.content)
        fragmentsList = ArrayList()
        fragmentsList!!.add(Fragment2_vp_1())
        fragmentsList!!.add(Fragment2_vp_2())
        fragmentsList!!.add(Fragment2_vp_3())
        fragmentsList!!.add(Fragment2_vp_4())
        /**
         * 实例化一个PagerAdapter
         * 必须重写的两个方法
         * getCount
         * getItem
         */
        val pagerAdapter: PagerAdapter = object : FragmentPagerAdapter(childFragmentManager) {
            override fun getItem(i: Int): Fragment {
                return fragmentsList!![i]
            }

            override fun getCount(): Int {
                return fragmentsList!!.size
            }
        }
        viewPager.setAdapter(pagerAdapter)
    }

    override fun onFragmentFirstVisible() {
    }

    init {
        setFragmentDelegater(FragmentDelegater(this))
    }
}